rllib/offline/estimators/direct_method.py

```diff
@@ -96,102 +35,62 @@ def __init__(
             name: string to save OPE results under
             policy: Policy to evaluate.
             gamma: Discount factor of the environment.
-            q_model_type: Either "fqe" for Fitted Q-Evaluation
-                or "qreg" for Q-Regression, or a custom model that implements:
-                - `estimate_q(states,actions)`
-                - `estimate_v(states, action_probs)`
-            train_test_split_val: Split the batch into a training batch with
-            `train_test_split_val * n_episodes` episodes and an evaluation batch
-            with `(1 - train_test_split_val) * n_episodes` episodes. If not
-            specified, use `k` for k-fold cross validation instead.
-            k: k-fold cross validation for training model and evaluating OPE.
-            kwargs: Optional arguments for the specified Q model.
+            q_model_config: Arguments to specify the Q-model.
         """
 
-        super().__init__(name, policy, gamma)
-        # TODO (rohan): Add support for continuous action spaces
-        assert isinstance(
-            policy.action_space, Discrete
-        ), "DM Estimator only supports discrete action spaces!"
         assert (
-            policy.config["batch_mode"] == "complete_episodes"
-        ), "DM Estimator only supports `batch_mode`=`complete_episodes`"
-
-        # TODO (rohan): Add support for TF!
-        if policy.framework == "torch":
-            if q_model_type == "qreg":
-                model_cls = QRegTorchModel
-            elif q_model_type == "fqe":
-                model_cls = FQETorchModel
-            else:
-                assert hasattr(
-                    q_model_type, "estimate_q"
-                ), "q_model_type must implement `estimate_q`!"
-                assert hasattr(
-                    q_model_type, "estimate_v"
-                ), "q_model_type must implement `estimate_v`!"
-        else:
-            raise ValueError(
-                f"{self.__class__.__name__}"
-                "estimator only supports `policy.framework`=`torch`"
-            )
+            policy.config["framework"] == "torch"
+        ), "DirectMethod estimator only works with torch!"
+        super().__init__(name, policy, gamma)
 
+        model_cls = q_model_config.pop("type")
         self.model = model_cls(
             policy=policy,
             gamma=gamma,
-            **kwargs,
+            **q_model_config,
         )
-        self.train_test_split_val = train_test_split_val
-        self.k = k
-        self.losses = []
+        assert hasattr(
+            self.model, "estimate_v"
+        ), "self.model must implement `estimate_v`!"
 
     @override(OffPolicyEstimator)
-    def estimate(self, batch: SampleBatchType) -> OffPolicyEstimate:
+    def estimate(self, batch: SampleBatchType) -> Dict[str, Any]:
```

Comment:
**todo**: don't forget the documentation, specifically on what you return and what they mean.